### ISSUE TITLE
Redis 3.4.0 results in error with flower and celery_worker

### ISSUE THREAD

**Issue by sirheimer** [pydanny/cookiecutter-django#2425]
## Error:

```
celeryworker_1  | [2020-01-31 16:06:46,914: CRITICAL/MainProcess] Unrecoverable error: TypeError("unhashable type: 'Redis'")
celeryworker_1  | Traceback (most recent call last):
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/worker/worker.py", line 205, in start
celeryworker_1  |     self.blueprint.start(self)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 119, in start
celeryworker_1  |     step.start(parent)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 369, in start
celeryworker_1  |     return self.obj.start()
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/consumer.py", line 318, in start
celeryworker_1  |     blueprint.start(self)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/bootsteps.py", line 119, in start
celeryworker_1  |     step.start(parent)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/mingle.py", line 40, in start
celeryworker_1  |     self.sync(c)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/mingle.py", line 44, in sync
celeryworker_1  |     replies = self.send_hello(c)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/worker/consumer/mingle.py", line 57, in send_hello
celeryworker_1  |     replies = inspect.hello(c.hostname, our_revoked._data) or {}
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/app/control.py", line 154, in hello
celeryworker_1  |     return self._request('hello', from_node=from_node, revoked=revoked)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/app/control.py", line 106, in _request
celeryworker_1  |     pattern=self.pattern, matcher=self.matcher,
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/celery/app/control.py", line 477, in broadcast
celeryworker_1  |     limit, callback, channel=channel,
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/pidbox.py", line 352, in _broadcast
celeryworker_1  |     channel=chan)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/pidbox.py", line 391, in _collect
celeryworker_1  |     self.connection.drain_events(timeout=timeout)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/connection.py", line 323, in drain_events
celeryworker_1  |     return self.transport.drain_events(self.connection, **kwargs)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/transport/virtual/base.py", line 963, in drain_events
celeryworker_1  |     get(self._deliver, timeout=timeout)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/transport/redis.py", line 374, in get
celeryworker_1  |     self._register_BRPOP(channel)
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/transport/redis.py", line 311, in _register_BRPOP
celeryworker_1  |     if not self._client_registered(channel, channel.client, 'BRPOP'):
celeryworker_1  |   File "/usr/local/lib/python3.7/site-packages/kombu/transport/redis.py", line 306, in _client_registered
celeryworker_1  |     (channel, client, cmd) in self._chan_to_sock)
celeryworker_1  | TypeError: unhashable type: 'Redis'
```

**Comment by sirheimer** [comment]
Error fixed when downgrading to version 3.3.11

**Comment by browniebroke** [comment]
Right, issue with Kombu:

https://github.com/celery/kombu/issues/1152
https://github.com/celery/kombu/issues/1153
